guides/source/action_text_overview.md

```diff
@@ -163,24 +146,66 @@ To customize the HTML rendered for embedded images and other attachments (known
+- [ffmpeg](http://ffmpeg.org/) v3.4+ for video previews and ffprobe for video/audio analysis
+- [poppler](https://poppler.freedesktop.org/) or [muPDF](https://mupdf.com/) for PDF previews
+
+You should familiarize yourself with the licensing implications of installing the above. Some, but not all of these libraries are required and they are dependent on the kind of uploads you are expecting within the editor. A common error that users encounter when working with Action Text and Active Storage is that images do not render correctly in the editor. This is usually due to the `libvips` dependency not being installed.
+
+
+### Signed GlobalId
+
+In addition to attachments uploaded through Active Storage, Action Text can also embed anything that can be resolved by a Signed GlobalID.
+
+A Global ID is an app wide URI that uniquely identifies a model instance: `gid://YourApp/Some::Model/id`. This is helpful when you need a single identifier to reference different classes of objects.
+
+When using this method, ActionText requires attachments to have a signed global ID (sgid). By default, all ActiveRecord models in a Rails app mix-in `GlobalID::Identification` concern. So they can be resolved by a signed global ID and are therefore `ActionText::Attachable` compatible.
 
-In addition to attachments uploaded through Active Storage, Action Text can
-embed anything that can be resolved by a [Signed
-GlobalID](https://github.com/rails/globalid#signed-global-ids).
+Action Text minifies the HTML you insert on save, so that it can re-render it with up-to-date content later on. This makes it so that you can reference models and always display the current content when those records change.
 
-Action Text renders embedded `<action-text-attachment>` elements by resolving
-their `sgid` attribute into an instance. Once resolved, that instance is passed
-along to
-[`render`](https://api.rubyonrails.org/classes/ActionView/Helpers/RenderingHelper.html#method-i-render).
-The resulting HTML is embedded as a descendant of the `<action-text-attachment>`
-element.
+ActionText will load up the model from the global id and then will render it with the default partial path when you render the content.
 
-For example, consider a `User` model:
+An Action Text Attachment can look like this:
+
+```html
+<action-text-attachment sgid="BAh7CEkiCG…"></action-text-attachment>
+```
+
+Action Text renders embedded `<action-text-attachment>` elements by resolving their sgid attribute of the element into an instance. Once resolved, that instance is passed along to a render helper. As a result the  HTML is embedded as a descendant of the `<action-text-attachment>` element.
+
+In order to be rendered within Action Text `<action-text-attachment>` element as an attachement, we must include the `ActionText::Attachable` module `implement #to_sgid(**options)` (made available through the `GlobalID::Identification` concern).
```

Comment:
Looks like the formatting got a bit strange on this paragraph, maybe `implement` wasn't supposed to be in code style?

<img width="659" alt="Screenshot 2024-02-05 at 15 19 56" src="https://github.com/Ridhwana/rails/assets/26328/c3c95f85-21b7-47d2-bd63-fad3bd48ca87">

```suggestion
In order to be rendered within Action Text `<action-text-attachment>` element as an attachment, we must include the `ActionText::Attachable` module, which implements `#to_sgid(**options)` (made available through the `GlobalID::Identification` concern).
```